drivers/counter/counter_ll_stm32_rtc.c

```diff
@@ -0,0 +1,344 @@
+
+	now.tm_hour = __LL_RTC_CONVERT_BCD2BIN(__LL_RTC_GET_HOUR(rtc_time));
+	now.tm_min = __LL_RTC_CONVERT_BCD2BIN(__LL_RTC_GET_MINUTE(rtc_time));
+	now.tm_sec = __LL_RTC_CONVERT_BCD2BIN(__LL_RTC_GET_SECOND(rtc_time));
+
+	ts = mktime(&now);
+
+	/* Return number of seconds since RTC init */
+	ts -= T_TIME_OFFSET;
+
+	ticks = counter_us_to_ticks(dev, (u64_t)(ts * USEC_PER_SEC));
+
+	return ticks;
+}
+
+static int rtc_stm32_set_alarm(struct device *dev, u8_t chan_id,
+				const struct counter_alarm_cfg *alarm_cfg)
+{
+	struct tm alarm_tm;
+	time_t alarm_val;
+	LL_RTC_AlarmTypeDef rtc_alarm;
+	struct counter_alarm_cfg *config = DEV_DATA(dev)->alarm_cfg;
+
+	u32_t now = rtc_stm32_read(dev);
+	u32_t ticks = alarm_cfg->ticks;
+
+	config->callback = alarm_cfg->callback;
+	config->user_data = alarm_cfg->user_data;
+
+	if (!alarm_cfg->absolute) {
+		ticks += now;
+	}
+
+	LOG_DBG("Set Alarm: %d\n", ticks);
+
+	alarm_val = (time_t)(counter_ticks_to_us(dev, ticks) / USEC_PER_SEC);
+
+	gmtime_r(&alarm_val, &alarm_tm);
+
+	/* Apply ALARM_A */
+	rtc_alarm.AlarmTime.TimeFormat = LL_RTC_TIME_FORMAT_AM_OR_24;
+	rtc_alarm.AlarmTime.Hours = alarm_tm.tm_hour;
+	rtc_alarm.AlarmTime.Minutes = alarm_tm.tm_min;
+	rtc_alarm.AlarmTime.Seconds = alarm_tm.tm_sec;
+
+	rtc_alarm.AlarmMask = LL_RTC_ALMA_MASK_NONE;
+	rtc_alarm.AlarmDateWeekDaySel = LL_RTC_ALMA_DATEWEEKDAYSEL_DATE;
+	rtc_alarm.AlarmDateWeekDay = alarm_tm.tm_mday;
+
+	LL_RTC_DisableWriteProtection(RTC);
+	LL_RTC_ALMA_Disable(RTC);
+	LL_RTC_EnableWriteProtection(RTC);
+
+	if (LL_RTC_ALMA_Init(RTC, LL_RTC_FORMAT_BIN, &rtc_alarm) != SUCCESS) {
+		return -EIO;
+	}
+
+	LL_RTC_DisableWriteProtection(RTC);
+	LL_RTC_ALMA_Enable(RTC);
+	LL_RTC_ClearFlag_ALRA(RTC);
+	LL_RTC_EnableIT_ALRA(RTC);
+	LL_RTC_EnableWriteProtection(RTC);
+
+	return 0;
+}
+
+
+static int rtc_stm32_disable_alarm(struct device *dev, u8_t chan_id)
+{
+	LL_RTC_DisableWriteProtection(RTC);
+	LL_RTC_ClearFlag_ALRA(RTC);
+	LL_RTC_DisableIT_ALRA(RTC);
+	LL_RTC_ALMA_Disable(RTC);
+	LL_RTC_EnableWriteProtection(RTC);
+
+	DEV_DATA(dev)->alarm_cfg = NULL;
+
+	return 0;
+}
+
+
+static u32_t rtc_stm32_get_pending_int(struct device *dev)
+{
+	return LL_RTC_IsActiveFlag_ALRA(RTC) != 0;
+}
+
+
+static u32_t rtc_stm32_get_top_value(struct device *dev)
+{
+	const struct counter_config_info *info = dev->config->config_info;
+
+	return info->max_top_value;
+}
+
+
+static int rtc_stm32_set_top_value(struct device *dev, u32_t ticks,
+				counter_top_callback_t callback,
+				void *user_data)
+{
+	ARG_UNUSED(dev);
+	ARG_UNUSED(ticks);
+	ARG_UNUSED(callback);
+	ARG_UNUSED(user_data);
+
+	return -ENOTSUP;
+}
+
+
+static u32_t rtc_stm32_get_max_relative_alarm(struct device *dev)
+{
+	const struct counter_config_info *info = dev->config->config_info;
+
+	return info->max_top_value;
+}
+
+
+void rtc_stm32_isr(void *arg)
+{
+	struct device *const dev = (struct device *)arg;
+	const struct counter_alarm_cfg *alarm_cfg = DEV_DATA(dev)->alarm_cfg;
```

Comment:
i don't see any place where `alarm_cfg` pointer would be set.

Reply:
right, good catch txs